jupyter/src/incremental_learning/job.py

```diff
@@ -200,6 +200,30 @@ def get_predictions(self) -> np.array:
                                    ['ml']['{}_prediction'.format(self.dependent_variable)])
         return np.array(predictions)
 
+    def get_probabilities(self) -> Union[None, np.array]:
+        """Return prediction probabilities for the class "true".
+
+        Note: Only works for binary classification.
+
+        Returns:
+            Union[None, np.array]: prediction probabilities. 
+        """
+        if self.analysis_name == 'regression':
+            logger.warning(
+                "Failed to obtain prediction probabilities for the regression job {}."
+                .format(self.name))
+            return None
+        probabilities = []
+        for item in self.results:
+            if 'row_results' in item:
+                target_prediction = item['row_results']['results']['ml']['{}_prediction'.format(
+                    self.dependent_variable)]
+                prediction_probability = item['row_results']['results']['ml']['prediction_probability']
+                if target_prediction == 'false':
```

Comment:
Does this work? `target_prediction` should be the class label in the input data. I would be inclined to use `{class_name: probability}`. Again we could cache the classes in `self` so we can get at it. Alternatively, this isn't an issue if we read from `top_classes`.

Reply:
I rewrote the code to augment `num_top_classes` into configuration and then use it for working with prediction probability.